gnomad_qc/v3/create_release/create_hgdp_tgp_subset.py

```diff
@@ -0,0 +1,821 @@
+            ht,
+            non_par=False,
+            prob_regions={"lcr": lcr_intervals.ht(), "segdup": seg_dup_intervals.ht()},
+        ),
+        allele_info=filters.allele_data,
+        **analyst_ht[ht.key],
+    )
+
+    return ht
+
+
+def hom_alt_depletion_fix(
+    mt: hl.MatrixTable,
+    af_expr: hl.expr.Float32Expression,
+    af_cutoff: float = 0.01,
+    ab_cutoff: float = 0.9,
+):
+    """
+    Adjust MT genotypes with temporary fix for the depletion of homozygous alternate genotypes.
+
+    :param mt: Input MT to add hom alt depletion genotype fix to
+    :param af_expr: Allele frequency expression to determine variants that need the hom alt fix
+    :param af_cutoff: Allele frequency cutoff for variants that need the hom alt fix
+    :param ab_cutoff: Allele balance cutoff to determine which genotypes need the hom alt fix
+    :return: MatrixTable with genotypes adjusted for the hom alt depletion fix
+    """
+    return mt.annotate_entries(
+        GT=hl.cond(
+            (af_expr > af_cutoff) & mt.GT.is_het() & ((mt.AD[1] / mt.DP) > ab_cutoff),
+            hl.call(1, 1),
+            mt.GT,
+        )
+    )
+
+
+def adjust_subset_alleles(mt: hl.MatrixTable):
+    """
+    Modeled after Hail's `filter_alleles` module
+
+    :param mt:
+    :return:
+    """
+    mt = mt.annotate_rows(
+        _keep_allele=hl.agg.array_agg(
+            lambda i: hl.agg.any(mt.LA.contains(i[0])), hl.enumerate(mt.alleles)
+        )
+    )
+    new_to_old = (
+        hl.enumerate(mt._keep_allele).filter(lambda elt: elt[1]).map(lambda elt: elt[0])
+    )
+    old_to_new_dict = hl.dict(
+        hl.enumerate(
+            hl.enumerate(mt.alleles).filter(lambda elt: mt._keep_allele[elt[0]])
+        ).map(lambda elt: (elt[1][1], elt[0]))
+    )
+    mt = mt.annotate_rows(
+        _old_to_new=hl.bind(
+            lambda d: mt.alleles.map(lambda a: d.get(a)), old_to_new_dict
+        ),
+        _new_to_old=new_to_old,
+    )
+    new_locus_alleles = hl.min_rep(
+        mt.locus, mt._new_to_old.map(lambda i: mt.alleles[i])
+    )
+    mt = mt.key_rows_by(
+        locus=new_locus_alleles.locus, alleles=new_locus_alleles.alleles
+    )
+    mt = mt.annotate_entries(LA=mt.LA.map(lambda x: mt._old_to_new[x]))
+
+    return mt.drop("_keep_allele", "_new_to_old", "_old_to_new")
+
+
+def create_full_subset_dense_mt(mt: hl.MatrixTable, meta_ht: hl.Table):
+    """
+    Create the subset dense release MatrixTable with multi-allelic variants and all sample and variant annotations.
+
+    .. note::
+
+        This function uses the sparse subset MT and assumes that centromeres and telomeres have already been filtered
+        if necessary.
+
+    :param mt: Sparse subset release MatrixTable
+    :param meta_ht: Metadata HT to use for sample (column) annotations
+    :return: Dense release MatrixTable with all row, column, and global annotations
+    """
+    release_ht = release(public=False).ht()
+    subset_freq = get_freq(subset="hgdp_tgp").ht()
```

Comment:
Yeah, as written it would need to be changed to take any other subsets. I think for now I would like to get the v3.1 code it, and then we can think of changes to make the subsetting more flexible, but happy to have further discussions about it.